### PR TITLE
Stub 24 hour view

### PR DESCRIPTION
**Pull request by kayernyc**
Sorry for the noisy commit - mostly thinking through the relatively simple case.

### CODE REVIEW

**Comment by deadnancy** [comment]
Hi, Kathrin! I saw `tslint` in package.json, but no config file. I generated one with `tslint --init`, then added `"lint": "tslint --project \"tsconfig.json\""` to your npm scripts. Running that resulted in a bunch of errors ([tslint_output.txt](https://github.com/kayernyc/configurable-date-picker/files/5022439/tslint_output.txt)), mainly `ban-types`, `no-string-literal`, and `no-trailing-whitespace`. Do you have a custom tsconfig.json you use? Or are these valid?

**Comment by kayernyc** [comment]
yes you are right - I need to add linting. I will open a new branch and rebase that in.

File: src/views/DatePickerListView.ts [hunk 1]
```diff
@@ -9,22 +13,39 @@ export default class DatePickerListView extends DatePickerBaseView
...
-      el.innerHTML = date.viewString
-      containerView.appendChild(el)
-    })
+  protected populateView(
+    model: DatePickerFactory,
+    containerView: HTMLElement,
+    frameElement: HTMLElement
+  ) {
+    throw new Error("Override populateView");
+  }
```
Comment (deadnancy) [inline src/views/DatePickerListView.ts:30]: Assuming this is just in progress, but why an error? Also curious why this is protected while its counterpart below in 12-hour view isn't. Mostly just confessing my confusion with TypeScript I think...

File: src/views/VirtualDom.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      initializeListeners();
+    } else {
+      console.log("WTF");
+    }
+  }
+
+  deallocate() {
+    // clean up all listeners
+  }
+}
```
Comment (deadnancy) [inline src/views/VirtualDom.ts:127]: The entire virtual DOM with scroll handling feels very reinvent-the-wheel. Assuming I'm missing something like this is part of an effort to create your own in-house vanilla JS framework from scratch...?

File: src/views/DatePickerListView.ts (hunk 1 above)
Comment (kayernyc) [reply src/views/DatePickerListView.ts:30]: Great question! This function must be overridden by the class that extends DatePickerListView, so this error forces anyone (future me) to actively consider what populateView should do _per_ implementation. It's a convention most commonly seen with Abstract classes.

Comment (deadnancy) [reply src/views/DatePickerListView.ts:30]: Thanks! I kind of figured out _what_ TypeScript abstract classes do in [pull request #5](https://github.com/kayernyc/configurable-date-picker/pull/5), but I was still unsure about _why_ you'd use them.

File: src/views/VirtualDom.ts (hunk 2 above)
Comment (kayernyc) [reply src/views/VirtualDom.ts:127]: exactly, trying to avoid pulling in anything external, and for this, a tiny, minimal dom is all that's necessary.

**Pull request merged**